go/runtime/nodes/grpc/grpc.go

```diff
@@ -435,61 +447,119 @@ func (cc *committeeClient) worker(ctx context.Context, ch <-chan *NodeUpdate, su
 			return
 		case <-rotCh:
 			// Local TLS certificates have been rotated, we need to refresh connections.
-			cc.logger.Debug("TLS certificates have been rotated, refreshing connections")
+			nc.logger.Debug("TLS certificates have been rotated, refreshing connections")
 
 			func() {
-				cc.Lock()
-				defer cc.Unlock()
+				nc.Lock()
+				defer nc.Unlock()
 
-				for id := range cc.conns {
-					cc.refreshConnectionLocked(id)
+				for id := range nc.conns {
+					nc.refreshConnectionLocked(id)
 				}
 			}()
 		case u := <-ch:
+			nc.logger.Debug("node update event received", "event", u)
 			func() {
-				cc.Lock()
-				defer cc.Unlock()
+				nc.Lock()
+				defer nc.Unlock()
 
 				switch {
 				case u.Reset:
-					// Committee has been reset.
-					for id := range cc.conns {
-						cc.deleteConnectionLocked(id)
+					// Node watcher reset.
+					for id := range nc.conns {
+						nc.deleteConnectionLocked(id)
 					}
 				case u.Freeze != nil:
-					// Committee has been frozen.
+					if !nc.nw.Versioned() {
+						nc.logger.Warn("versioned event for non-versioned node watcher",
+							"event", u,
+						)
+						return
+					}
```

Comment:
Will this cause the client to silently stop functioning? This seems like a developer error, maybe it deserves a panic?

Reply:
It won't stop functioning since the `return` here is only to exit the closure (which is used so we can `defer` unlock).

But yeah this indeed is a developer error so I was also thinking about panicking here instead.

Reply:
I'm not sure if we should panic here or not. On one hand we probably should since this should never happen unless it's a development error, on the other hand if this error does happen, this will generally still work ok.

Probably should panic nonetheless.

Reply:
I'll leave it as a warning for now